File: docs/content/fundamentals/fast-io/code/fast_io.v1.full.cpp
/**
 * Fast I/O — read many integers quickly and print their sum.
 *
 * `ios::sync_with_stdio(false)` + `cin.tie(nullptr)` unhook the C++ streams from
 * C stdio; on large inputs this is dramatically faster and avoids Time Limit
 * Exceeded.
 *
 * Input:  an integer n, then n integers.
 * Output: the sum of the n integers.
 */
#include <iostream>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int n;
    cin >> n;
    long long sum = 0;
    for (int i = 0; i < n; i++) {
        int x;
        cin >> x;
        sum += x;
    }
    cout << sum << "\n";
    return 0;
}

File: docs/algorithms/io/fast-io/code/fast_io.v1.full.py
"""Fast I/O — read many integers quickly and print their sum.

Reading the whole input at once with `sys.stdin.buffer.read()` is much faster
than calling `input()` in a loop, which avoids Time Limit Exceeded on big cases.

Input:  an integer n, then n integers.
Output: the sum of the n integers.
"""
import sys


def main() -> None:
    data: list[bytes] = sys.stdin.buffer.read().split()
    n: int = int(data[0])
    total: int = sum(int(x) for x in data[1:1 + n])
    print(total)


main()
